### PR TITLE
[call-hierarchy] Implement proposed LSP extension (WIP)

### PR DESCRIPTION
**Pull request by AlexTugarev**
This PR implements the client side of the [updated LSP proposal](https://github.com/Microsoft/vscode-languageserver-node/pull/420#issuecomment-440288747) for a call hierarchy. 

screen shot 2019-01-23 at 16 41 55

* related PR for the `typescript-language-server`: https://github.com/theia-ide/typescript-language-server/pull/98
* related PR for the LSP extension: https://github.com/Microsoft/vscode-languageserver-node/pull/420

**WIP** because the PR for `typescript-language-server` isn't merged yet.

### CODE REVIEW

File: packages/typescript/src/node/typescript-contribution.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ export class TypeScriptContribution extends BaseLanguageServerContribution {
         if (isRequestMessage(message)) {
             if (message.method === InitializeRequest.type.method) {
                 const initializeParams = message.params as TypeScriptInitializeParams;
+                const capabilities = initializeParams.capabilities;
+                if (capabilities.textDocument) {
+                    if (!capabilities.textDocument.documentSymbol) {
+                        capabilities.textDocument.documentSymbol = {};
+                    }
+                    capabilities.textDocument.documentSymbol.hierarchicalDocumentSymbolSupport = true;
```
Comment (AlexTugarev) [inline packages/typescript/src/node/typescript-contribution.ts]: @akosyakov, do you know why this capability isn't `true` by default in Theia?

**Comment by kittaakos** [comment]
Can we (re-)use the `tabbar-toolbar` contribution to have

 - a toggle button for `Caller/Calle` and 
 - two commands: `Refresh`, and `Cancel`

in the widgets? (Same for the type hierarchy UI; `Subtype/Supertype`.)

screen shot 2019-01-28 at 22 24 24

File: packages/typescript/src/node/typescript-contribution.ts (hunk 1 above)
Comment (akosyakov) [reply packages/typescript/src/node/typescript-contribution.ts]: it should be: https://github.com/TypeFox/vscode-languageserver-node/blob/fb80c5e2db4c714032c28d6f482e770642ae9e23/client/src/client.ts#L1574

Comment (AlexTugarev) [reply packages/typescript/src/node/typescript-contribution.ts]: well, this isn't the case. I'm investigating, why `capabilities.textDocument.documentSymbol` is not present at all.

Comment (AlexTugarev) [reply packages/typescript/src/node/typescript-contribution.ts]: thanks for the pointer, BTW!

Comment (AlexTugarev) [reply packages/typescript/src/node/typescript-contribution.ts]: nvm, I had a type in `fillClientCapabilities`.

**Comment by AlexTugarev** [comment]
@kittaakos, I've updated the PR with latest changes to the protocol (proposed.)

**Comment by AlexTugarev** [comment]
https://github.com/Microsoft/vscode-languageserver-node/pull/420 is merged! so back to square one an reimplement this :-)

**Comment by vince-fugnitto** [comment]
@AlexTugarev is this pull-request closeable thanks to #6924 ?

**Comment by AlexTugarev** [comment]
'closing as rotten.

**Pull request closed without merging**